cmd/worker/worker.go

```diff
@@ -0,0 +1,80 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package worker
+
+import (
+	"context"
+	"fmt"
+	"sync"
+
+	"github.com/Azure/acr-cli/cmd/api"
+)
+
+// PurgeWorker defines a worker that can process PurgeJobs.
+type PurgeWorker struct {
+	Job         chan PurgeJob
+	WorkerQueue chan chan PurgeJob
+	StopChan    chan bool
+	wg          *sync.WaitGroup
+}
+
+// NewPurgeWorker creates a new worker.
+func NewPurgeWorker(wg *sync.WaitGroup, workerQueue chan chan PurgeJob) PurgeWorker {
+	worker := PurgeWorker{
+		Job:         make(chan PurgeJob),
+		WorkerQueue: workerQueue,
+		wg:          wg,
+	}
+	return worker
+}
+
+// Start starts a new purgeWork with an infinite loop inside a goroutine.
+func (pw *PurgeWorker) Start() {
+	go func() {
+		for {
+			// Free worker, insert worker into worker queue.
+			WorkerQueue <- pw.Job
+
+			select {
+			// If the worker has a job assigned begin processing it.
+			case job := <-pw.Job:
+				ctx := context.Background()
+				pw.ProcessJob(ctx, job)
+			// If the worker needs to be stopped then return.
+			case <-pw.StopChan:
+				return
+			}
+		}
+	}()
+}
+
+// Stop notifies the worker to stop handling purge jobs
+func (pw *PurgeWorker) Stop() {
+	go func() {
+		pw.StopChan <- true
+	}()
+}
+
+// ProcessJob processes any job (currently PurgeTag and PurgeManifest)
+func (pw *PurgeWorker) ProcessJob(ctx context.Context, job PurgeJob) {
+	go func() {
+		defer pw.wg.Done()
+		switch job.JobType {
+		case PurgeTag:
+			err := api.AcrDeleteTag(ctx, job.LoginURL, job.Auth, job.RepoName, job.Tag)
+			if err != nil {
```

Comment:
you could put a `var err error` outside of the case and then share the error logic